cmd/snapd-apparmor/main.go

```diff
@@ -0,0 +1,149 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// This script is provided for integration with systemd on distributions where
+// apparmor profiles generated and managed by snapd are not loaded by the
+// system-wide apparmor systemd integration on early boot-up.
+//
+// Only the start operation is provided as all other activity is managed by
+// snapd as a part of the life-cycle of particular snaps.
+//
+// In addition the script assumes that the system-wide apparmor service has
+// already executed, initializing apparmor file-systems as necessary.
+//
+// NOTE: This script ignores failures in some scenarios as the intent is
+// to simply load application profiles, as many as we can, even if for
+// whatever reason some of those fail.
```

Comment:
It's stil not clear to me why failures are not treated as errors. Is it because this script is just a mean to speed-up the execution of snaps by pre-loading their profile? If so, I'd add a "for performance purposes" somewhere in this sentence.

Reply:
Thanks will do.